packages/story-editor/src/components/panels/design/pageBackground/test/pageBackground.js

```diff
@@ -0,0 +1,135 @@
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { fireEvent, screen } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import PageBackgroundPanel from '../pageBackground';
+import { renderPanel } from '../../../shared/test/_utils';
+import ConfigContext from '../../../../../app/config/context';
+import { StoryContext } from '../../../../../app/story';
+import useFFmpeg from '../../../../../app/media/utils/useFFmpeg';
+
+jest.mock('@web-stories-wp/design-system', () => ({
+  ...jest.requireActual('@web-stories-wp/design-system'),
+  useSnackbar: jest.fn(() => ({ showSnackbar: jest.fn() })),
+}));
+jest.mock('../../../../../app/media/utils/useFFmpeg');
+
+const mockUseFFmpeg = useFFmpeg;
+
+function MediaUpload({ render, onSelect }) {
+  const open = () => {
+    const image = { type: 'image', src: 'media1' };
+    onSelect(image);
+  };
+
+  return render(open);
+}
+
+function arrange(selectedElements) {
+  const configValue = {
+    allowedImageFileTypes: ['gif', 'jpe', 'jpeg', 'jpg', 'png'],
+    allowedFileTypes: ['gif', 'jpe', 'jpeg', 'jpg', 'png'],
+    allowedImageMimeTypes: [
+      'image/png',
+      'image/jpeg',
+      'image/jpg',
+      'image/gif',
+    ],
+    allowedMimeTypes: {
+      image: ['image/png', 'image/jpeg', 'image/jpg', 'image/gif'],
+      video: [],
+    },
+    capabilities: {
+      hasUploadMediaAction: true,
+    },
+    allowedTranscodableMimeTypes: [],
+    MediaUpload,
+  };
+
+  const combineElements = jest.fn();
+  const storyValue = {
+    state: {
+      currentPage: {
+        backgroundColor: {},
+      },
+    },
+    actions: {
+      clearBackgroundElement: jest.fn(),
+      combineElements,
+      updateCurrentPageProperties: jest.fn(),
+    },
+  };
+
+  const wrapper = ({ children }) => (
+    <ConfigContext.Provider value={configValue}>
+      <StoryContext.Provider value={storyValue}>
+        {children}
+      </StoryContext.Provider>
+    </ConfigContext.Provider>
+  );
+
+  return {
+    ...renderPanel(PageBackgroundPanel, selectedElements, wrapper),
+    combineElements,
+  };
+}
+
+describe('Panels/PageBackground', () => {
+  beforeAll(() => {
+    localStorage.setItem(
+      'web_stories_ui_panel_settings:pageBackground',
+      JSON.stringify({ isCollapsed: false })
+    );
+  });
+
+  afterAll(() => {
+    localStorage.clear();
+  });
+
+  it('should simulate a click on the replace button', () => {
+    mockUseFFmpeg.mockReturnValue({ isTranscodingEnabled: false });
+    const backgroundImage = {
+      x: 0,
+      y: 0,
+      width: 1,
+      height: 1,
+      id: '123',
+      isBackground: true,
+      type: 'image',
+      rotationAngle: 0,
+      resource: {
+        id: 'resource/123',
+        src: 'media.jpg',
+        alt: 'Media',
+      },
+    };
+    const { combineElements } = arrange([backgroundImage]);
+    const replaceButton = screen.getByRole('button', { name: 'Replace' });
```

Comment:
This ought to be moved to `fixture.editor.inspector.designPanel.pageBackground.replace`.

Reply:
This is a unit test though, not Karma.